diff --git a/lib/Utils/messages-media.js b/lib/Utils/messages-media.js
index a3d2fa6bae6af021a4a50b4270584fed08623f34..c3f3ba4d1a4278280baac43d09274321e6dfd362 100644
--- a/lib/Utils/messages-media.js
+++ b/lib/Utils/messages-media.js
@@ -5,7 +5,7 @@ import { once } from 'events';
 import { createReadStream, createWriteStream, promises as fs, WriteStream } from 'fs';
 import { tmpdir } from 'os';
 import { join } from 'path';
-import { Readable, Transform } from 'stream';
+import { Readable, Transform, pipeline } from 'stream';
 import { URL } from 'url';
 import { proto } from '../../WAProto/index.js';
 import { DEFAULT_ORIGIN, MEDIA_HKDF_KEY_MAPPING, MEDIA_PATH_MAP } from '../Defaults/index.js';
@@ -297,6 +297,8 @@ export async function generateThumbnail(file, mediaType, options) {
 export const getHttpStream = async (url, options = {}) => {
     const response = await fetch(url.toString(), {
         dispatcher: options.dispatcher,
+        signal: options.signal,
+        redirect: options.redirect,
         method: 'GET',
         headers: options.headers
     });
@@ -512,7 +514,8 @@ export const downloadEncryptedContent = async (downloadUrl, { cipherKey, iv }, {
             }
         }
     });
-    return fetched.pipe(output, { end: true });
+    pipeline(fetched, output, () => {});
+    return output;
 };
 export function extensionForMediaMessage(message) {
     const getExtension = (mimetype) => mimetype.split(';')[0]?.split('/')[1];
